app/_lib/pdfstitcher.ts

```diff
@@ -28,16 +28,18 @@ function trimmedPageSize(
 ) {
   /**
    * Computes the size for each trimmed page.
-   * Assumes that all selected pages are the same!
+   * Chooses the largest page width and height from the user specified page range to match how the pdf viewer works.
    */
-  const firstRealPage = (pages.find((p) => p > 0) || 1) - 1;
-  const firstPage = inDoc.getPage(firstRealPage);
-
-  const pageSize = firstPage.getTrimBox() || firstPage.getMediaBox();
-  const width = pageSize.width - settings.edgeInsets.horizontal;
-  const height = pageSize.height - settings.edgeInsets.vertical;
+  let width = 0;
+  let height = 0;
+  for (const page of pages) {
```

Comment:
I would suggest filtering for pages > 0 in case someone adds a blank space, otherwise `getPage` throws an error.

Reply:
Ooo, yes, silly me. Thank you!